src/model/entities/rundown.ts

```diff
@@ -701,10 +701,26 @@ export class Rundown extends BasicRundown {
     this.assertActive(this.insertPartAsNext.name)
     this.assertNotUndefined(this.activeCursor, 'active Segment')
 
+    this.updateRankFromOnAirAndNextParts(part)
     this.activeCursor.segment.insertPartAfterActivePart(part)
     this.setNext(this.activeCursor.segment.id, part.id)
   }
 
+  private updateRankFromOnAirAndNextParts(partToBeUpdated: Part): void {
```

Comment:
A solution can be to find the closest rank (if any) after the on air part and take find a rank in between. If there are no parts after the on air part in the segment then the "found" rank, then the on air rank + 1 can be used.
Pseudocode:
```typescript
private updateRankFromOnAirPart(part: Part): void {
  if (!this.activeCursor) {
    return
  }
  const onAirRank: number = this.activeCursor.part.getRank()
  const onAirPartIndex: number = this.activeCursor.segment.parts.findIndex(part => part.id === this.activeCursor.part.id)
  if (onAirPartIndex < 0) {
    return
  }
  const nextTakenRank: number = this.activeCursor.segment.parts[onAirPartIndex + 1]?.rank ?? (onAirRank + 1)
  const partRank: number = (nextTakenRank - onAirRank) / 2 + onAirRank
  part.updateRank(partRank)
}
```

Reply:
Update with a bit more clean approach (based on your suggestion):
```typescript
private updateRankFromOnAirPart(part: Part): void {
  if (!this.activeCursor) {
    return
  }
  const onAirRank: number = this.activeCursor.part.getRank()
  const nextTakenRank: number = this.getNextTakenPartRank(this.activeCursor.part)
  const partRank: number = (nextTakenRank - onAirRank) / 2 + onAirRank
  part.updateRank(partRank)
}

private getNextTakenPartRank(part: Part): number {
  try {
    return this.getPartAfter(part).rank
  } catch {
    return part.rank + 1
  }
}
```

Reply:
I have updated the calculation.
I am not using your suggestion since manipulating the Parts of the Segments from the Rundown is dancing on the border of the law of demeter.

Reply:
`getPartAfter` returns the next valid Part to be taken. This Part _could_ be on another Segment. So it's quite possible that the rank of the onAir Part is higher than `nextTakenRank`.

Reply:
I assume that you are referring to the first suggestion with the comment about the law of demeter.
Per my second comment, I suggest that getting the rank is made its own responsibility by separating it, so the catch doesn't have to handle potential failures of the other logic in the update rank method.
The name of the method should be changed, since it no longer uses the next part.
If `0.1` is kept, then it would be nice to have a descriptive constant of what that value represents.

Reply:
Method name has been updated.

GitHub hadn't updated, so when I posted my first comment on this thread, I could only see your first comment.